xarray/core/duck_array_ops.py

```diff
@@ -171,6 +171,79 @@ def _ignore_warnings_if(condition):
         yield
 
 
+def _nansum(value, axis=None, **kwargs):
+    """ In house nansum for object array """
+    value = fillna(value, 0.0)
+    return _dask_or_eager_func('sum')(value, axis=axis, **kwargs)
+
+
+def _nan_minmax(func, fill_value, value, axis=None, **kwargs):
+    """ In house nanmin and nanmax for object array """
+    valid_count = count(value, axis=axis)
+    filled_value = fillna(value, fill_value)
+    data = _dask_or_eager_func(func)(filled_value, axis=axis, **kwargs)
+    if not hasattr(data, 'dtype'):  # scalar case
+        data = np.nan if valid_count == 0 else data
+        return np.array(data, dtype=value.dtype)
+    return where_method(data, valid_count != 0)
+
+
+def _nan_argminmax(func, fill_value, value, axis=None, **kwargs):
+    """ In house nanargmin, nanargmax for object arrays. Always return integer
+    type """
+    valid_count = count(value, axis=axis)
+    value = fillna(value, fill_value)
+    data = _dask_or_eager_func(func)(value, axis=axis, **kwargs)
+    # dask seems return non-integer type
+    if isinstance(value, dask_array_type):
+        data = data.astype(int)
+    if not hasattr(data, 'dtype'):  # scalar case
+        # TODO should we raise ValueError if all-nan slice encountered?
+        data = -1 if valid_count == 0 else int(data)
+        return np.array(data)
+    return where_method(data, valid_count != 0, -1)
+
+
+def _nanmean_ddof(ddof, value, axis=None, **kwargs):
+    """ In house nanmean. ddof argument will be used in _nanvar method """
+    valid_count = count(value, axis=axis)
+    value = fillna(value, 0.0)
+    # TODO numpy's mean does not support object-type array, so we assume float
+    dtype = kwargs.get('dtype', None)
+    if dtype is None and value.dtype.kind == 'O':
+        dtype = value.dtype if value.dtype.kind in ['cf'] else float
+    data = _dask_or_eager_func('mean')(value, axis=axis, dtype=dtype, **kwargs)
+
+    # adjust the sample size
+    if axis is None:
+        size = value.size
+    else:
+        size = np.prod(value.shape[axis])
+    data = data / (valid_count - ddof) * size
+    return where_method(data, valid_count != 0)
+
+
+def _nanvar(value, axis=None, **kwargs):
+    ddof = kwargs.pop('ddof', 0)
+    kwargs_mean = kwargs.copy()
+    kwargs_mean.pop('keepdims', None)
+    value_mean = _nanmean_ddof(ddof=0, value=value, axis=axis, keepdims=True,
+                               **kwargs_mean)
+    squared = _dask_or_eager_func('square')(value.astype(value_mean.dtype) -
```

Comment:
You could potentially just use the operator `** 2` instead of the dask_or_eager_func here.

Reply:
Done